addons/project/security/project_security.xml

```diff
@@ -231,6 +231,28 @@
         <field name="groups" eval="[(4,ref('base.group_user'))]"/>
     </record>
 
+    <record model="ir.rule" id="report_project_task_user_rule">
+        <field name="name">Tasks Analysis: project visibility User</field>
+        <field name="model_id" ref="model_report_project_task_user"/>
+        <field name="domain_force">[
+        '|',
+            ('project_id.privacy_visibility', '!=', 'followers'),
+            '|',
+                ('project_id.message_partner_ids', 'in', [user.partner_id.id]),
+                '|',
+                    ('task_id.message_partner_ids', 'in', [user.partner_id.id]),
+                    ('user_ids', 'in', user.id)
+        ]</field>
+        <field name="groups" eval="[(4,ref('base.group_user'))]"/>
```

Comment:
Since the report is only available to the project user. We could set the project user group instead of this group. 🙂